cpp/src/transpose/transpose.cu

```diff
@@ -46,17 +70,17 @@ void gpu_transpose_valids(gdf_valid_type **in_cols_valid,
   gdf_size_type stride_x = blockDim.x * gridDim.x;
   gdf_size_type stride_y = blockDim.y * gridDim.y;
 
-  MaskType* const __restrict__ out_mask32 =
-    reinterpret_cast<MaskType*>(out_cols_valid[y]);
-
   for(gdf_size_type i = x; i < ncols; i += stride_x)
   {
     for(gdf_size_type j = y; j < nrows; j += stride_y)
     {
-      auto active_threads = __ballot_sync(0xffffffff, i < ncols && j < nrows);
+      auto active_threads = __ballot_sync(0xffffffff, 1);
```

Comment:
This fix still isn't correct. You are still trying to sync all threads (mask 0xffffffff), even though some may not make it into this loop.

Reply:
This wasn't my attempted fix. This was just a cleanup after thinking that i don't really need to fix it. @jrhemstad proved with wrong because even though it was working, it was undefined behaviour.